File: LeetCode/0650. 2 Keys Keyboard/solution.go
/**
 * 126 / 126 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 1.9 MB
 */
func minSteps(n int) int {
	if n == 1 {
		return 0
	}
	x := n >> 1
	for ; n%x != 0; x-- {}
	return minSteps(x) + (n / x)
}

File: LeetCode/0650. 2 Keys Keyboard/solution.py
"""
126 / 126 test cases passed.
Runtime: 28 ms
Memory Usage: 15 MB
"""
class Solution:
    def minSteps(self, n: int) -> int:
        if n == 1:
            return 0
        x = n >> 1
        while n % x != 0:
            x -= 1
        return self.minSteps(x) + (n // x)
